mcsema/Arch/ABI.cpp

```diff
@@ -342,10 +345,11 @@ static const char *IntReturnValVar(llvm::CallingConv::ID cc) {
 }
 
 static const char *FloatReturnValVar(llvm::CallingConv::ID cc,
-                                     llvm::Type *type) {
+                                     llvm::Type *type, size_t index) {
   if (llvm::CallingConv::X86_64_SysV == cc ||
       llvm::CallingConv::Win64 == cc) {
-    return "XMM0";
+    static const char *regs[2] = {"XMM0", "XMM1"};
+    return regs[index];
 
   } else if (llvm::CallingConv::X86_StdCall == cc ||
```

Comment:
Another similar thing here relating to non-zero indexes.

Also, though it's not in code you touched, we should probably handle the case of `type->isDoubleType()` for `gArch->IsX86()` under the `C` calling convention. Not sure what's right to do there.